### DISCUSSION TITLE
Is it possible to have port forwarding/a public URL without needing to login?

### DISCUSSION THREAD

**Discussion by marks** [community/community#27195] [Codespaces]
<p>I am trying to create an integration that requires a platform’s web service to hit my endpoint. I am developing the endpoint code (Flask app) within a Github Codespace. I am able to visit the URL from my browser (<a href="https://xyz-3000.apps.codespaces.githubusercontent.com/endpoint-abc" rel="noopener nofollow ugc">https://xyz-3000.apps.codespaces.githubusercontent.com/endpoint-abc</a>) where I am logged into Github but when I try in an incognito window or using <code>curl</code>, I get redirected to a Github login page.</p>
<p>Is there any way to bypass this or is this by design to not allow people to host services, even in development, on Codespaces?</p>
<p>Many thanks!</p>

**Comment by lostintangent** [comment, marked as answer]
<p>Hey! Adding support for public/unauthenticated ports is on our roadmap, but it isn’t currently supported. We’ll make sure to share out updates as we make progress on this :+1:</p>

**Comment by marks** [comment]
<p>Thanks for the response! Glad it’s not just me. Might be worth making it a little more explicit in the <a href="https://docs.github.com/en/free-pro-team@latest/github/developing-online-with-codespaces/developing-in-a-codespace#forwarding-ports" rel="noopener nofollow ugc">docs</a>.</p>
<p>By the way… any concerns with running something like <a href="https://ngrok.com/" rel="noopener nofollow ugc">ngrok</a> in a codespace, to get around this current limitation?</p>

**Comment by lostintangent** [comment]
<p>Yeah using ngrok would be a good near-term workaround for this!</p>

**Comment by marks** [comment]
<p>Responding to myself to say that running ngrok inside a Codespace worked just fine</p>
